### PR TITLE
pixel-shift-combiner 1.5.0,1050,sgwn5e23 (new cask)

### PR DESCRIPTION
**Pull request by yunhao-jiang**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---
1. The `url` is quite long because of the weird versioning by Fujifilm (1.5.0 -> ...1050.dmg), have to `rjust` the minor version so that it match. Don't know if there's a better way of doing it, is it allowed to introduce variable to the cask file?
2. I added this cask to auto-bump. Is there a rule of what should be added and what shouldn't? This doesn't update that frequent, is it ok to add to auto-bump?
3. The last line of the zap trash contained a weird hidden unicode character in the path ([U+2068]). This cannot be removed or zap won't work just FYI. -- Solved by using `*`

Thanks!

### CODE REVIEW

**Comment by bevanjkay** [comment]
>     The url is quite long because of the weird versioning by Fujifilm (1.5.0 -> ...1050.dmg), have to rjust the minor version so that it match. Don't know if there's a better way of doing it, is it allowed to introduce variable to the cask file?

This may the only way to handle it, especially if it remains consistently in the same format across version updates.

>     I added this cask to auto-bump. Is there a rule of what should be added and what shouldn't? This doesn't update that frequent, is it ok to add to auto-bump?

There isn't a rule, it is ok to add as long as it doesn't use the `extract_plist` strategy.

>     There is last line of the zap trash contained a weird hidden unicode character in the path ([U+2068]). This cannot be removed or zap won't work just FYI.

You could try put a glob `*` in place of where the character should be?

File: Casks/p/pixel-shift-combiner.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      "#{match[1]},#{match[2]}"
+    end
+  end
+
+  depends_on macos: ">= :high_sierra"
+
+  app "Pixel Shift Combiner.app"
+
+  zap trash: ["~/Library/Preferences/com.fujifilm.denji.PIXEL-SHIFT-COMBINER.plist",
+              "~/Library/Application Support/com.fujifilm.denji/*Pixel Shift Combiner"]
```
Comment (bevanjkay) [inline Casks/p/pixel-shift-combiner.rb:26]:
```suggestion
  zap trash: [
    "~/Library/Preferences/com.fujifilm.denji.PIXEL-SHIFT-COMBINER.plist",
    "~/Library/Application Support/com.fujifilm.denji/*Pixel Shift Combiner",
  ]

Comment (yunhao-jiang) [reply Casks/p/pixel-shift-combiner.rb]: Thanks! Weird that `brew style --fix` didn't suggest this to me. I'll make sure keep a record of this.

File: Casks/p/pixel-shift-combiner.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  livecheck do
+    url :homepage
+    regex(%r{Mac\sVersion:\s(\d+(?:\.\d+)+).*href=.*?pixel-shift-combiner-mac\d+-([a-zA-Z0-9]+)/}im)
+    strategy :page_match do |page, regex|
+      match = page.match(regex)
+      next if match.blank?
+
+      "#{match[1]},#{match[2]}"
+    end
```
Comment (khipp) [inline Casks/p/pixel-shift-combiner.rb:18]:
```suggestion
    regex(%r{Mac\sVersion:\s(\d+(?:\.\d+)+).*href=.*?pixel-shift-combiner-mac(\d+)[._-]([a-zA-Z0-9]+)/}im)
    strategy :page_match do |page, regex|
      match = page.match(regex)
      next if match.blank?

      "#{match[1]},#{match[2]},#{match[3]}"
    end
```

File: Casks/p/pixel-shift-combiner.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+cask "pixel-shift-combiner" do
+  version "1.5.0,sgwn5e23"
+  sha256 "4e734f66062e19c7272abd06e6d21b5bdc222090631e2691ce8b847975e5fce1"
+
+  url "https://dl.fujifilm-x.com/support/software/pixel-shift-combiner-mac#{version.csv.first.major}#{version.csv.first.minor.rjust(2, "0")}#{version.csv.first.patch}-#{version.csv.second}/FUJIFILM_PixelShiftCombiner#{version.csv.first.major}#{version.csv.first.minor.rjust(2, "0")}#{version.csv.first.patch}.dmg"
```
Comment (khipp) [inline Casks/p/pixel-shift-combiner.rb:5]:
```suggestion
  version "1.5.0,1050,sgwn5e23"
  sha256 "4e734f66062e19c7272abd06e6d21b5bdc222090631e2691ce8b847975e5fce1"

  url "https://dl.fujifilm-x.com/support/software/pixel-shift-combiner-mac#{version.csv.second}-#{version.csv.third}/FUJIFILM_PixelShiftCombiner#{version.csv.second}.dmg"
```

**Review by khipp — COMMENTED** [review]
> The `url` is quite long because of the weird versioning by Fujifilm (1.5.0 -> ...1050.dmg), have to rjust the minor version so that it match. Don't know if there's a better way of doing it, is it allowed to introduce variable to the cask file?

We could try to additionally capture that part inside the `livecheck`.

File: Casks/p/pixel-shift-combiner.rb (hunk 3 above)
Comment (yunhao-jiang) [reply Casks/p/pixel-shift-combiner.rb]: I think this way is a much safer and cleaner way no matter how Fujifilm decide to name their file. Thanks!

**Review by bevanjkay — APPROVED** [review]

**Pull request merged**